### ISSUE TITLE
ES6+ module

### ISSUE THREAD

**Issue by tomexx** [harc/ohm#185]
It would be great if ES6+ modules were supported and user could simply import the modules in a standard way. Any plans to do that? Thanks

**Comment by pdubroy** [comment]
Sorry, I could have sworn I replied to this right after you opened it!

I'd be happy to support importing `ohm` as an ES6 module if it's not too big of a change. I don't yet know enough about ES6 modules though -- can you suggest what we'd need to do to support this?

**Comment by jedahan** [comment]
I made an example of how I'd like to use ohm, that uses rollup to build stuff.... but it doesn't work. I am not familiar enough with how ohm is structured, or how commonjs / es6 modules work to fix the test case error. https://github.com/jedahan/requiring-ohm
```
(node:35387) ExperimentalWarning: The ESM module loader is experimental.
TypeError: Cannot read property 'PExpr' of undefined
   at /Volumes/data/code/requiring-ohm/build/room.js:1469:10
    at Failure.getPExpr (/Volumes/data/code/requiring-ohm/build/room.js:2:65)
    at Object.<anonymous> (/Volumes/data/code/requiring-ohm/build/room.js:5:2)
```

It seems Failure is being inlined, so there is no `this` to refer to

**Comment by pdubroy** [comment]
I've tested `import`ing Ohm in Node, and it seems to work. So I'm going to close this, but please re-open if there is something specific that is not working.

@jedahan, I do not know enough about Rollup to be able to help here. If there is something specific we can change that would help, please open a new issue and I'll see what I can do.

**Comment by 7185** [comment]
I have the following message in my Angular project:
```
...ActionParser.js depends on 'ohm-js'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
```
It seems ohm-js is not following the ES module standard, so maybe this issue could be reopened?

**Comment by pdubroy** [comment]
@7185 I'm working on a fix in #350. Could you please try install `ohm-js@16.2.0-pre.esm` and let me know if it fixes the problem?

**Comment by 7185** [comment]
@pdubroy It seems to fix the problem indeed, no more warning with that version

**Comment by pdubroy** [comment]
@7185 Thanks for confirming!

I've just published `ohm-js@16.2.0` with this.
